src/datasets/summarizations/trend-partial.summarization.ts

```diff
@@ -0,0 +1,103 @@
+import * as math from 'mathjs';
+import { Summary } from './types';
+import { TimeSeriesPoint } from '../metas/types';
+import { cacheSummaries } from './utils/commons';
+import {
+  PointMembershipFunction,
+  MembershipFunction,
+  trapmf,
+  trapmfL,
+  trapmfR,
+} from './libs/protoform';
+import {
+  exponentialMovingAverage,
+  normalizedUniformPartiallyLinearEpsApprox,
+  TimeSeriesPartialTrend,
+} from './libs/trend';
+import { formatX, formatY } from '../../utils/formatters';
+
+export function queryFactory(points: TimeSeriesPoint[]) {

```

Comment:
Add a comment to describe what a function does. It's especially useful for these longer functions. A more specific name would also help. Our style guide requires descriptions for all exported functions.

Reply:
This function will be removed later in refactor-summarization PR